SCORER.md

```diff
@@ -21,11 +28,278 @@ You may need to build your own scorer - your own _language model_ if:
+
+In the following example we will create a custom external scorer file for Bahasa Indonesia (BCP47: `id-ID`).
+
+#### Preparing the text file
+
+This is straightforward. In this example, we will use a file called `indonesian-sentences.txt`. This file should contain phrases that you wish to prioritize recognising. For example, you may want to recognise place names, digits or medical phrases - and you will include these phrases in the `.txt` file.
+
+_These phrases should not be copied from `test.tsv`, `train.tsv` or `validated.tsv` as you will bias the resultant model._
+
+```
+~/deepspeech-data$ ls cv-corpus-6.1-2020-12-11/id
+total 6288
+   4 drwxr-xr-x 3 root root    4096 Feb 24 19:01 ./
+   4 drwxr-xr-x 4 root root    4096 Feb 11 07:09 ../
+1600 drwxr-xr-x 2 root root 1638400 Feb  9 10:43 clips/
+ 396 -rwxr-xr-x 1 root root  401601 Feb  9 10:43 dev.tsv
+ 104 -rwxr-xr-x 1 root root  103332 Feb  9 10:43 invalidated.tsv
+1448 -rwxr-xr-x 1 root root 1481571 Feb  9 10:43 other.tsv
+  28 -rwxr-xr-x 1 root root   26394 Feb  9 10:43 reported.tsv
+ 392 -rwxr-xr-x 1 root root  399790 Feb  9 10:43 test.tsv
+ 456 -rwxr-xr-x 1 root root  465258 Feb  9 10:43 train.tsv
+1848 -rwxr-xr-x 1 root root 1889606 Feb  9 10:43 validated.tsv
+```
+
+The `indonesian-sentences.txt` file is stored on the local filesystem in the `deepspeech-data` directory so that the Docker container can access it.
+
+```
+~/deepspeech-data$ ls | grep indonesian-sentences
+ 476 -rw-rw-r--  1 root root  483481 Feb 24 19:02 indonesian-sentences.txt
+```
+
+The `indonesian-sentences.txt` file is formatted with one phrase per line, eg:
+
+```
+Kamar adik laki-laki saya lebih sempit daripada kamar saya.
+Ayah akan membunuhku.
+Ini pulpen.
+Akira pandai bermain tenis.
+Dia keluar dari ruangan tanpa mengatakan sepatah kata pun.
+Besok Anda akan bertemu dengan siapa.
+Aku mengerti maksudmu.
+Tolong lepas jasmu.
+```
+
+#### Using `lm_optimizer.py` to generate values for the parameters `--default_alpha` and `--default_beta` that are used by the `generate_scorer_package` script
+
+The `lm_optimizer.py` script is located in the `DeepSpeech` directory if you have set up your [environment][ENVIRONMENT.md] as outlined in the PlayBook.
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# ls | grep lm_optimizer.py
+lm_optimizer.py
+```
+
+This script takes a set of test data (`--test_files`), and a `--checkpoint_dir` parameter and determines the optimal `--default_alpha` and `--default_beta` values.
+
+Call `lm_optimizer.py` and pass it the `--test_files` and a `--checkpoint_dir` directory.
+
+```
+root@57e6bf4eeb1c:/DeepSpeech# python3 lm_optimizer.py \
+     --test_files deepspeech-data/cv-corpus-6.1-2020-12-11/id/clips/test.csv \
+     --checkpoint_dir deepspeech-data/checkpoints
+```
+
+Remember, if you trained your model with a `--n_hidden` value that is different to the default (`1024`), you should pass the same `--n_hidden` value to `lm_optimizer.py`, for example:
```

Comment:
nit: in general, any change to the default geometry needs to be reflected otherwise checkpoint will fail to load

Reply:
Good call.